compiler/damlc/daml-ide/src/DA/Daml/LanguageServer/Visualize.hs

```diff
@@ -0,0 +1,56 @@
+-- Copyright (c) 2019 The DAML Authors. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- | Execute Daml commands
+
+module DA.Daml.LanguageServer.Visualize
+    ( setCommandHandler
+    ) where
+
+import qualified Data.Aeson as Aeson
+import           Language.Haskell.LSP.Types
+import           Development.IDE.LSP.Server
+
+import Development.IDE.Types.Logger
+
+import qualified Data.Text as T
+
+import Development.IDE.Core.Rules
+import Development.IDE.Core.Service.Daml
+
+import Development.IDE.Core.Shake
+import Development.IDE.Core.RuleTypes.Daml
+
+import Language.Haskell.LSP.Messages
+import qualified Language.Haskell.LSP.Core as LSP
+import Development.IDE.Types.Location
+
+collectTexts :: List Aeson.Value -> Maybe NormalizedFilePath
+collectTexts (List [Aeson.String file])  = Just (toNormalizedFilePath (T.unpack file))
+collectTexts _= Nothing
+
+onCommand
+    :: IdeState
+    -> ExecuteCommandParams
+    -> IO Aeson.Value
+onCommand ide execParsms = case execParsms of
+    ExecuteCommandParams "daml/damlVisualize" (Just _arguments) -> do
+        case collectTexts _arguments of
+            Just mod -> do
+                    logInfo (ideLogger ide) "Generating visualization for current daml project"
+                    Just dots <- runAction ide (use GenerateVisualization mod)
+                    return $ Aeson.String dots
+            Nothing     -> do
+                logError (ideLogger ide) "Expected a single module to visualize, got multiple module"
+                return $ Aeson.String "Expected a single module to visualize, got multiple module"
+    ExecuteCommandParams  _ (Just _arguments) -> do
```

Comment:
Command handlers are expected to be extensible. If you get commands that you cannot handle, just ignore them.

Reply:
`PartialHandlers` is expected to be extensible but I think we will have to handle all the ExecuteCommands once we have `ExecuteCommandHandler`. I am not sure how we can ignore without getting incomplete match warning

Reply:
Good point, we cannot handle commands that partially handle a request at the moment. That will need some refactoring. I would suggest that you combine the two cases here (the first where you have arguments but the wrong command and the second where you don’t have arguments into a single case and log both the command name and the arguments.

Reply:
Cool. Will do that